### PR TITLE
Chat & Builder apps

### PR DESCRIPTION
**Pull request by sfc-gh-amiribel**
# Summary 

This PR begins the revamp of the current admin app experience. We are deprecating the current combined app and first starting with a standalone "chat" app, which is intended for users to iterate and add verified queries to an existing semantic model.

Eventually, we will reincorporate the from-scratch builder flow, but for now we are starting with the iteration step.

Follow [the branch README.md](https://github.com/Snowflake-Labs/semantic-model-generator/blob/c997edac7877233f1c25e0a32265b89cc8e2d318/admin_apps/README.md) to get started and try those apps.

# To-dos

- [x] **Common 'chrome' for both apps.** Maybe there should be some light and consistent chrome on those two apps just to make them look coherent. For now, the builder app has logo and title, chat app has nothing. Maybe there's a middle ground!
- [x] **Make sure the env. variables instructions are clear.** In both apps, we kinda require that env variables are properly set. It's stressed in the repo's README, laybe we should add more instructions in this README too.
- [x] **Make sure the back-end actually works.** I think the two apps have the UI almost done - but there's a significant part of plugging wires that must be done. Supposedly, all utils should be found in the main package, or already in the `shared_utils.py`. 
- [x] **Separate Streamlit apps requirements from main package.** For now, the app requirements are being consolidated in the `pyproject.toml` which really is about the semantic-model-generator package, and, to me, should not include the underlying streamlit apps requirements. 
- [ ] **Update apps requirements.** I think the current stable streamlit build is currently missing `streamlit-monaco` (the code editor) and could almost afford to switch from `streamlit-nightly` to `streamlit==1.36.0` once ready.
- [ ] **Add tests.** Streamlit has introduced a way to [test Streamlit apps](https://docs.streamlit.io/develop/api-reference/app-testing), that could be a neat way to ensure we don't break things in the app!

### CODE REVIEW

File: admin_apps/vqr_app.py
Comment (sfc-gh-amiribel) [inline admin_apps/vqr_app.py:1]: This is not functional for now. I hit an API problem. To just work on the flow, I made `vqr_app_mock.py` instead

File: admin_apps/vqr_app_mock.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
+from __future__ import annotations
+import time
+from typing import Literal
+import pandas as pd
+import streamlit as st
+from dataclasses import dataclass
+
+import streamlit as st
+from streamlit_monaco import st_monaco
```
Comment (sfc-gh-amiribel) [inline admin_apps/vqr_app_mock.py:9]: This `streamlit_monaco` will need to be added to the poetry env if we like it better than simple st.text_area

File: admin_apps/yaml_app_utils/pages/edit.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+from shared_utils import edit_semantic_model, semantic_model_exists
+
+st.session_state["next_is_unlocked"] = True
+
+if semantic_model_exists():
+    edit_semantic_model()
+else:
+    st.error("No model found.")
+
+# st.caption("Prototypes:")
```
Comment (sfc-gh-amiribel) [inline admin_apps/yaml_app_utils/pages/edit.py:11]: (This and below is just ugly WIP to make cards as shown earlier)

File: admin_apps/vqr_app.py [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+init_session_states()
+
+connector = SnowflakeConnector(
+    account_name=SNOWFLAKE_ACCOUNT,
+    max_workers=1,
+)
+
+
+def get_file_name() -> str:
```
Comment (sfc-gh-rehuang) [inline admin_apps/vqr_app.py:59]: @arnaudmiribel you should update here to take in explicit file name input

**Comment by sfc-gh-rehuang** [comment]
@sfc-gh-cnivera I pushed a small instruction update but still running into package issue & probably easier to just do req.txt for admin app. 
Also the chat isn't working as expected; seems something todo with the api endpoint update? Or am i missing something
Screenshot 2024-07-11 at 1 54 54 PM

**Review by sfc-gh-rehuang — APPROVED** [review]
Looks really good! Thank you so much for all the changes!

**Pull request merged**